src/ncp/ncp_base.cpp

```diff
@@ -5900,6 +5944,62 @@ ThreadError NcpBase::InsertPropertyHandler_MAC_WHITELIST(uint8_t header, spinel_
     return errorCode;
 }
 
+#if OPENTHREAD_ENABLE_COMMISSIONER
+ThreadError NcpBase::InsertPropertyHandler_SPINEL_PROP_THREAD_JOINERS(uint8_t header, spinel_prop_key_t key,
+                                                                      const uint8_t *value_ptr, uint16_t value_len)
+{
+    spinel_ssize_t parsedLength;
+    ThreadError errorCode = kThreadError_None;
+
+    otExtAddress *ot_ext_address = NULL;
+    const char *aPSKd = NULL;
+    const int PSK_MIN_LENGTH = 6;
+    const int PSK_MAX_LENGTH = 32;
+    int psk_len;
+
+    VerifyOrExit(
+        mAllowLocalNetworkDataChange == true,
+        errorCode = SendLastStatus(header, SPINEL_STATUS_INVALID_STATE)
+    );
+
+    parsedLength = spinel_datatype_unpack(
+                       value_ptr,
+                       value_len,
+                       "EDc",
```

Comment:
This should be changed from `"EDc"` to `"ED."`¹. Eventually, you will be able to change this to just be `"Ed"`.

Explanation: I (regrettably) defined `'D'` to behave differently depending on if it was the last type in the string or not. If `'D'` was the last type in the string, then it would omit the data length (assuming it to be redundant) and assume that the rest of the bytes in the buffer belonged to that field. If `'D'` was not the last type in the string, a length would be prepended to the data so that parsing could continue after the type.

What I should have done is define two separate types: one with a length, and one without. This would always make it explicit which one was being used. And that's what I intend to do with `'d'`: it will be the same as `'D'` except that it always prepends the length, no matter where it appears in the string. This will help us avoid all of this `'D.'` madness².

In the mean time, please use `"ED."` here.

Sorry for that confusion.

-----------

¹ If the type signature for this property was `"A(T(ED))"` (note, without a `"."` after the D), then you would parse this struct with `"ED"` and it would have parsed just fine. The reason I asked you to change it to `"A(T(ED.))"` and `"ED."` (Note the `"."`) was to ensure that we could append additional datatypes later on without breaking backward compatibility.

² `"T(...)"` actually has a similar problem to `"D"`, since they are somewhat related. Will be introducing a `"t(...)"` type as well, with the same placement-independent behavior as `"D"`.